### ISSUE TITLE
Warning texts are printed as if it was concurrent

### ISSUE THREAD

**Issue by hanjoosten** [commercialhaskell/stack#4059]
For some reason, cache became out of date. When building with stack build, is looks like multiple concurrent processes print a warning message. Together, this becomes unreadable.

### Expected

I'd expect that warning messages are human readable.

### Actual

~~~~
PS C:\Users\hjo20125\Git\Ampersand> stack build
WWAARNIRNNGI:N Gc:a cchaec hies  iosu to uotf  odfa tdea:t eC:: /CU:s/eUrsse/rhsj/oh20j1o2250/1A2p5p/DAaptpaD/aLtoac/aLlo/caPlr/oPgrroagmrsa/msst/asctka/cxk8/6x_8664_-6w4i-nwdionwdso/wgsh/cg-h8c.-08..20\.l2i\bl\ipba\cpkaacgkea.gceo.ncfo.ndf\.pda\cpkaacgkea.gcea.cchaec
hgeh
cg hwci lwli lsle es eaen  aonl do lvdi evwi eowf  otfh itsh ipsa cpkaacgkea gdeb .d bU.s eU s'eg h'cg-hpck-gp krge craecchaec'h et'o  tfoi xf.i
x.
WARNING: cache is out of date: C:/Users/hjo20125/AppData/Local/Programs/stack/x86_64-windows/ghc-8.0.2\lib\package.conf.d\package.cache
ghc will see an old view of this package db. Use 'ghc-pkg recache' to fix.
ampersand-3.10.0: unregistering (local file changes: src\Ampersand\Basics\BuildInfo_Generated.hs)
ampersand-3.10.0: configure (lib + exe)
Configuring ampersand-3.10.0...
ampersand-3.10.0: build (lib + exe)
~~~~

### Stack version

```
PS C:\Users\hjo20125\Git\Ampersand> stack --version
Version 1.7.1, Git revision 681c800873816c022739ca7ed14755e85a579565 (5807 commits) x86_64 hpack-0.28.2
```
I am on Windows 10.

**Comment by mgsloan** [comment]
It may be worth trying `stack upgrade --git` -with the merge of
https://github.com/commercialhaskell/stack/pull/4028 , it no longer does.
If that doesn't resolve it, then it is probably coming from some other
process invocation other than ghc-pkg, which inherits stderr.

**Comment by snoyberg** [comment]
This should in fact be closed by #4028, please ping again if it's still affecting you with `master`.

**Comment by hanjoosten** [comment]
Confirmed. Thanks.
